README.Rmd

```diff
@@ -0,0 +1,191 @@
+[r2dii.match](https://github.com/2DegreesInvesting/r2dii.match), and
+[r2dii.analysis](https://github.com/2DegreesInvesting/r2dii.analysis).
+
+You may want to analyze a single portfolio and investor, or multiple ones. This document details two workflows:
+
+* __Single-inputs workflow__: When the number of portfolios and investors is
+exactly one, we use a specific set of scripts optimized for this purpose. This
+workflow is typically used to generate interactive reports. Because this
+workflow is available [online](https://platform.transitionmonitor.com/start), it
+has been referred to as "the online workflow", and the scripts as "the web tool
+scripts". This is misleading because this workflow can run offline too.
+* __Multiple-outputs workflow__: When the number of portfolios and/or investors
+is not exactly one, we use a different set of optimized scripts. This workflow
+is typically used to produce static (.pdf) reports. By contrast with the
+so-called "online" workflow, this one has been usually referred to as "the
+offline workflow"; again, this is misleading.
+
+Both workflows include three steps:
+
+1. Clean portfolio: Clean the input portfolio and merge in financial data, to
+categorize each holding and identify whether it's equity (EQ) or corporate bonds
+(CB).
+2. Run Analysis: Merge portfolios with asset-level data and scenarios, then
+group results at company, portfolio and regional level.
+3. Present results: Present the results in a clear output format.
+
+--
+
+Note: Code chunks with prompts "`$`" and "`#>`" correspond to bash and R,
+respectively.
+
+### Siblings
+
+PACTA_analysis depends on other "siblings" of the PACTA family. They are all
+available at <https://github.com/2DegreesInvesting> and should live alongside
+PACTA_analysis/, under the same parent directory. All siblings include an empty file called ".pacta", which you can use to find them programmatically.
+
+```{bash comment="$"}
+siblings="$(dirname $(find .. -maxdepth 2 -empty -type f -name '.pacta'))"
+echo $siblings
+```
+
+Before any analysis, ensure all siblings are up to date.
+
+```{bash comment="$"}
+# Store
+ working_directory="$pwd"
+
+siblings="$(dirname $(find .. -maxdepth 2 -empty -type f -name '.pacta'))"
+for i in $siblings; do
+  cd $i && echo $i
+  git log --decorate --oneline -n 1
+done
+
+# Restore
+cd "$working_directory"
+
+```
+
+For an easier way to work with all siblings at once from the terminal see
+[pacta-cli](https://github.com/2DegreesInvesting/pacta-cli).
+
+## Computing environment
+
+The required computing environment is available in [this docker
+image](https://github.com/2DegreesInvesting/docker/tree/master/r-packages).
+
+Other computing environments may be defined as workflows for GitHub actions.
+
+```{bash comment="$"}
+ls .github/workflows
+```
+
+Unless otherwise indicated, R packages should be up to date.
+
+<details> <summary>Session information.</summary>
+```{r}
+devtools::session_info()
+```
+</details>
```

Comment:
not sure what this is for

Reply:
This exposes the computing environment (including the version of each package) by expanding a dropdown icon. 

https://github.com/2DegreesInvesting/PACTA_analysis/blob/generalize-workflow/README.md#computing-environment

![image](https://user-images.githubusercontent.com/5856545/109855401-11346e80-7c0d-11eb-9dbc-74949e6decc9.png)

Expanded

![image](https://user-images.githubusercontent.com/5856545/109855437-1bef0380-7c0d-11eb-9f05-5ce5d4d7bfbb.png)

Reply:
maybe a comment about why it's here... do you expect the user to do this? do you expect them to review the information in it? is there something particular they should be looking for or checking?